consult.el

```diff
@@ -76,10 +76,10 @@
   :type 'function)
 
 (defcustom consult-mode-histories
-  '((eshell-mode . eshell-history-ring)
-    (comint-mode . comint-input-ring)
-    (term-mode   . term-input-ring))
-  "Alist of (mode . history) pairs of mode histories.
+  '((eshell-mode . (eshell-history-ring eshell-history-index))
+    (comint-mode . (comint-input-ring comint-input-ring-index))
+    (term-mode   . (term-input-ring nil)))
+  "Alist of (mode . (history index)) pairs of mode histories.
```

Comment:
I would prefer lists here (eshell-mode eshell-history-ring eshell-history-index)

Reply:
Needs an update of the comment too.